src/controllers/auth/signup.ts

```diff
@@ -0,0 +1,19 @@
+import * as Interfaces from "@interfaces";
+import * as Utils from "@utils";
+
+const signup: Interfaces.Middlewares.Async = async (req, res) => {
+  const body: Interfaces.User = req.body;
+  const user = await Utils.Auth.createUser(body);
+  if (req.session) {
+    req.session.user = { uid: user.id, username: user.uname };
```

Comment:
Is there a need to populate req.session?

Reply:
[for this](https://github.com/expressjs/cookie-session#:~:text=The%20middleware%20will,for%20the%20session)